service/history/workflowTaskHandler.go

```diff
@@ -1423,49 +1411,68 @@ func (handler *workflowTaskHandlerImpl) validateCommandAttr(
 	validationFn commandAttrValidationFn,
 ) error {
 
-	if failedCause, err := validationFn(); err != nil {
-		if _, ok := err.(*serviceerror.InvalidArgument); ok {
-			return handler.failCommand(failedCause, err)
-		}
+	return handler.failWorkflowTaskOnInvalidArgument(validationFn())
+}
+
+func (handler *workflowTaskHandlerImpl) failWorkflowTaskOnInvalidArgument(
```

Comment:
why we need this method? can we just call failWorkflowTask(wtFailedCause, err) in places where you need this method?

Reply:
I don't want to change any logic in "refactor" style PR. But yes, having both of them is not right. I was thinking about reverse approach: calling `failWorkflowTaskOnInvalidArgument` everywhere where `failWorkflowTask` is called now. Wrong "user" input should fail WT but any server errors should not. I need to track all possible errors here to make a call and it is out of scope for this small refactoring.